tutor/src/models/tour.js

```diff
@@ -14,48 +14,45 @@ import User from './user';
 
 const TourInstances = new Map();
 
-function getTour(id, options = {}) {
-  const tourSettings = TourData[id];
-  const { courseId } = options;
-  let tourData = tourSettings;
-  let tourId = id;
-
-  if (courseId) {
-    if (tourSettings.perCourse) {
-      tourId = `${id}-${courseId}`;
-    }
-    tourData = defaults({ courseId }, tourSettings);
-  }
-
-  let tour = TourInstances.get(tourId);
-  if (!tour){
-    tour = new Tour(tourData);
-    TourInstances.set(tourId, tour);
-  }
-  return tour;
-}
-
 export default
 @identifiedBy('tour')
 class Tour extends BaseModel {
 
   static forIdentifier(id, options) {
-    return TourData[id] ? getTour(id, options) : undefined;
+    const tourSettings = TourData[id];
+    if (!tourSettings) {
+      return undefined;
+    }
+
+    const { courseId } = options;
+    let tourData;
+
+    if (courseId) {
+      if (tourSettings.perCourse) {
+        tourId = `${id}-${courseId}`;
+      }
+      tourData = defaults({ courseId }, tourSettings);
+    }
+    else {
+      if (tourSettings.perCourse) {
+        return undefined;
+      }
+      tourData = tourSettings;
+    }
+
+    let tourId = id;
+    let tour = TourInstances.get(tourId);
+    if (!tour){
+      tour = new Tour(tourData);
+      TourInstances.set(tourId, tour);
+    }
+    return tour;
   }
```

Comment:
you should declare `tourId` and the others at the top of the function.  You're currently declare it on line 43 but are assigning for the first time on 32.  A good practice is to declare all variables at the top of the function.

Reply:
Yes you're right I'll move the lets to the top.